modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -989,7 +989,7 @@ private CompletableFuture<List<BinaryRow>> processScanRetrieveBatchAction(ReadWr
             TableSchemaAwareIndexStorage indexStorage = secondaryIndexStorages.get().get(request.indexToUse());
 
             if (indexStorage == null) {
-                throw new AssertionError("Index not found: uuid=" + request.indexToUse());
+                throw new IllegalStateException("Index not found: uuid=" + request.indexToUse());
```

Comment:
Here and below. 
- According to our guidelines we should put all params in square braces `Index not found [uuid=" + request.indexToUse() +']'` or you may use ` IgniteStringFormatter.format(Index not found [uuid={}]`, request.indexToUse())
- Index id is actually an Integer and not uuid, thus I'd also rename uuid to just id.
- AFAIC According to our guidelines we should put a full stop at the end of the exception message.